source/local/local.go

```diff
@@ -89,22 +87,43 @@ func (ls *localSourceHandler) CacheKey(ctx context.Context, g session.Group, ind
 }
 
 func (ls *localSourceHandler) Snapshot(ctx context.Context, g session.Group) (cache.ImmutableRef, error) {
+	sessionID := ls.src.SessionID
+	if sessionID == "" {
+		return ls.snapshotWithAnySession(ctx, g)
+	}
+
+	timeoutCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	caller, err := ls.sm.Get(timeoutCtx, sessionID, false)
+	if err != nil {
+		return ls.snapshotWithAnySession(ctx, g)
+	}
+
+	ref, err := ls.snapshot(ctx, caller)
+	if err != nil {
+		if errors.Is(err, filesync.TransferNotStartedError{}) {
```

Comment:
Not sure I understand how this would work. You are comparing with an empty error?

Maybe just call it `InvalidSessionError`